### PR TITLE
Upgrade errorprone to enable java 11 builds

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/LongMeasure.java [hunk 1]
```diff
@@ -69,7 +69,7 @@
    * @since 0.1.0
    */
   @ThreadSafe
-  interface BoundLongMeasure extends BoundInstrument {
+  interface BoundLongMeasure extends InstrumentWithBinding.BoundInstrument {
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: Please send these changes in a different PR, they are javadoc warnings, saw them in the CI.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: Sure thing. Need to get the build working on CI, too.

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: Don't think for that small PR you need to fix anything :)

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: Without that fix, this change doesn't work, for reasons I don't understand.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: But I can do it first thing tomorrow.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/metrics/LongMeasure.java]: done!

**Comment by jkwatson** [comment]
I'm not sure why the "normal" java 8 builds was getting OOM killed so much, but it seems like restricting it to only using 1G of heap has fixed it.

**Comment by carlosalberto** [comment]
```
Callable expected <1> but was <1> within 5 seconds.
```

That's a rather weird message. And it _looks_ on a super exotic bug (on any of the involved components).

**Pull request merged**